### PR TITLE
Reset question block cache delete

### PR DESCRIPTION
**Pull request by Jokkis80**
Deletes the Cache to reset a Question Block

**GENERAL**
- Deletes Redis Cache for Question Blocks

### CODE REVIEW

File: src/services/sessionExec.js [hunk 1]
```diff
@@ -366,6 +366,41 @@ const joinSession = async ({ shortname }) => {
...
+const resetQuestionBlock = async ({ id, instanceIds }) => {
+  const session = await getRunningSession(id)
+  instanceIds.forEach(async instanceId => {
+    const type = await responseCache.hget(`instance:${instanceId}:info`, 'type')
+    const questionInstance = await QuestionInstanceModel.findOne({ _id: instanceId })
+    const question = await QuestionModel.findOne({ _id: questionInstance.question })
+    if (type === QUESTION_TYPES.SC || type === QUESTION_TYPES.MC) {
+      let maxChoices = 0
+      if (type === QUESTION_TYPES.SC) {
+        maxChoices = question.versions[question.__v].options.SC.choices.length
```
Comment (rschlaefli) [inline src/services/sessionExec.js]: I think MongoDB __v does not necessarily correlate with the Klicker question version. Would be better to use a version identifier that is stored in Info, or to check the length of the redis hash directly (hgetall minus one for the participants key).

Comment (rschlaefli) [reply src/services/sessionExec.js]: Also, if the length were calculate directly from redis, would not need to treat sc and mc questions differently.

Comment (rschlaefli) [reply src/services/sessionExec.js]: This could also save us the database lookup for each instance, as the type is already in redis info.

Comment (Jokkis80) [reply src/services/sessionExec.js]: Updated deleting loops and MongoDB request removed

File: src/services/sessionExec.js [hunk 2]
```diff
@@ -366,6 +366,40 @@ const joinSession = async ({ shortname }) => {
...
+      Object.keys(responseResults).forEach(key => {
+        responseCache.hset(`instance:${instanceId}:results`, `${key}`, 0)
+      })
+    }
+
+    if (type === QUESTION_TYPES.FREE || type === QUESTION_TYPES.FREE_RANGE) {
+      const responseHashes = await responseCache.hgetall(`instance:${instanceId}:responseHashes`)
+
+      Object.keys(responseHashes).forEach(key => {
+        responseCache.hdel(`instance:${instanceId}:responseHashes`, `${key}`)
```
Comment (rschlaefli) [inline src/services/sessionExec.js]: We might need to map this to promises and use await Promise.all() in the end.

File: src/services/sessionExec.js [hunk 3]
```diff
@@ -366,6 +366,40 @@ const joinSession = async ({ shortname }) => {
...
+      })
+
+      Object.keys(responseResults).forEach(key => {
+        responseCache.hdel(`instance:${instanceId}:results`, `${key}`)
+      })
+    }
+    await responseCache.hset(`instance:${instanceId}:results`, 'participants', 0)
```
Comment (rschlaefli) [inline src/services/sessionExec.js]: Same as above.

**Pull request merged**